modules/standard/IO.chpl

```diff
@@ -3865,6 +3866,58 @@ proc channel.readline(arg: [] uint(8), out numRead : int, start = arg.domain.low
   return false;
 }
 
+/*
+  Read a line into a Chapel array of bytes. Reads until a ``\n`` is reached.
+  This function always does a binary read (i.e. it is not aware of UTF-8 etc)
+  and is similar in some ways to `readLine(bytes)` but works with an array directly.
+  However, it does not resize the array but rather stores up to first
+  'size' bytes in to it. The exact number of bytes in the array set will depend
+  on the line length (and on stripNewline since the newline will be counted if it is
+  stored in the array).
+
+  :arg arg: A 1D DefaultRectangular array which must have at least 1 element.
+  :arg maxSize: The maximum amount of bytes to read.
+  :arg stripNewline: Whether to strip the trailing ``\n`` from the line.
+  :returns: Returns `0` if EOF is reached and no data is read. Otherwise, returns the number of array elements that were set by this call.
+
+  :throws SystemError: Thrown if data could not be read from the channel.
```

Comment:
Let's note that it throws if the line is longer than `maxSize` bytes. Also looking at the code I am not certain it does that. Can you check and add a test to lock it in?